### PR TITLE
edit: update all enforcer reports so that the models support storage + new query APIs

### PR DESCRIPTION
**Pull request by aaslamin**
Issue: https://github.com/aporeto-inc/aporeto-devs/issues/36

### CODE REVIEW

File: specs/flowreport.spec [hunk 1]
```diff
@@ -6,6 +6,23 @@ model:
...
+- - namespace
+  - timestamp
+- - timestamp
+- - action
+- - remoteNamespace
+- - observedAction
+- - dropReason
+- - destinationPort
+- - sourcePlatform
+- - destinationPlatform
```
Comment (aaslamin) [inline specs/flowreport.spec:25]: 👀 

I picked indices based on the selection made here: https://github.com/aporeto-inc/backend/blob/master/srv/zack/internal/processors/flowreport.go#L127-L134

Recall that `tags` result in an index being created in InfluxDB - https://docs.influxdata.com/influxdb/v1.7/concepts/key_concepts/#tag-key

File: specs/flowreport.spec [hunk 2]
```diff
@@ -6,6 +6,23 @@ model:
...
   description: Post a new flow log.
+  extends:
+  - '@identifiable-stored'
+  - '@zoned'
+  - '@migratable'
+
+# Indexes
+indexes:
+- - namespace
+  - timestamp
```
Comment (aaslamin) [inline specs/flowreport.spec:17]: The sole timestamp index will likely become our TTL index to prune old flow reports - https://docs.mongodb.com/manual/core/index-ttl/

Comment (aaslamin) [inline specs/flowreport.spec:17]: I am guessing this is likely to become our most useful index to handle range queries

File: alarm.go [hunk 3]
```diff
@@ -1353,8 +1349,6 @@ func (o SparseAlarmsList) List() elemental.IdentifiablesList {
 func (o SparseAlarmsList) DefaultOrder() []string {
 
 	return []string{
-		"name",
-		":no-inherit",
 		"updateTime",
 	}
 }
```
Comment (aaslamin) [inline alarm.go:1354]: ℹ️I am guessing the last commit merged to master did not include this update to the alarm model. I used the latest & greatest `rego` & `elegen` binaries to generate the code which likely caused this change.

File: specs/flowreport.spec [hunk 4]
```diff
@@ -6,6 +6,23 @@ model:
...
+- - timestamp
```
Comment (aaslamin) [inline specs/flowreport.spec:18]: The sole timestamp index will likely become our TTL index to prune old flow reports - https://docs.mongodb.com/manual/core/index-ttl/

File: specs/reportsquery.spec [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    - Audit
+    - Enforcers
+    - Files
+    - EventLogs
+    - Packets
+    - EnforcerTraces
+    - Counters
+    - Accesses
+    - DNSLookups
+    - PingReports
```
Comment (aaslamin) [inline specs/reportsquery.spec:71]: @primalmotion just confirming, but we want to *remove* `PingReports` as a queryable report type from this endpoint, correct?

**Comment by aaslamin** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "victoria-metrics",
    "component": "manipulate",
    "pr-id": "127",
    "commit-sha": "bf22db4122e4dcf62c73a2c107c8fe3d3c3cad72",
    "pipeline": "master"
  },
  {
    "project": "victoria-metrics",
    "component": "backend",
    "pr-id": "726",
    "commit-sha": "fe450be2aee3ffa59e1fe9d6a2d1ed4958c46e5e",
    "pipeline": "master"
  },
  {
    "project": "victoria-metrics",
    "component": "gaia",
    "pr-id": "471",
    "commit-sha": "8c87f216ac3978968e3d2ff087a3e4c60b60ae1f",
    "pipeline": "master"
  },
  {
    "project": "victoria-metrics",
    "component": "underwater",
    "pr-id": "104",
    "commit-sha": "52843e51e546536d79d893dffb53428d30cf8c5d",
    "pipeline": "master"
  }
]
```

**Comment by aaslamin** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "victoria-metrics",
    "component": "gaia",
    "pr-id": "471",
    "commit-sha": "ad180e323d772d94bdfbcd0f6bc91d9458ad5582",
    "pipeline": "master"
  },
  {
    "project": "victoria-metrics",
    "component": "backend",
    "pr-id": "726",
    "commit-sha": "53daf859fe9458adc537021e416344200560a6e2",
    "pipeline": "master"
  }
]
```

**Comment by aaslamin** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "victoria-metrics",
    "component": "backend",
    "pr-id": "726",
    "commit-sha": "5a4dbb464ed473b11ecda10eec64b04a2b232f70",
    "pipeline": "master"
  },
  {
    "project": "victoria-metrics",
    "component": "gaia",
    "pr-id": "471",
    "commit-sha": "ad180e323d772d94bdfbcd0f6bc91d9458ad5582",
    "pipeline": "master"
  }
]
```

**Comment by aaslamin** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "victoria-metrics",
    "component": "backend",
    "pr-id": "726",
    "commit-sha": "10cf5ae07b126bb3f6b5812ffb477ed02ce1053b",
    "pipeline": "master"
  },
  {
    "project": "victoria-metrics",
    "component": "gaia",
    "pr-id": "471",
    "commit-sha": "ad180e323d772d94bdfbcd0f6bc91d9458ad5582",
    "pipeline": "master"
  }
]
```

**Comment by aaslamin** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "victoria-metrics",
    "component": "backend",
    "pr-id": "726",
    "commit-sha": "eefbafddc974aa6ab505cc2b013b077d13857478",
    "pipeline": "master"
  },
  {
    "project": "victoria-metrics",
    "component": "gaia",
    "pr-id": "471",
    "commit-sha": "ad180e323d772d94bdfbcd0f6bc91d9458ad5582",
    "pipeline": "master"
  }
]
```

**Comment by aaslamin** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "victoria-metrics",
    "component": "backend",
    "pr-id": "726",
    "commit-sha": "7e997854d8a9c58bd658df9c836ce4d69749cf1b",
    "pipeline": "master"
  },
  {
    "project": "victoria-metrics",
    "component": "gaia",
    "pr-id": "471",
    "commit-sha": "ad180e323d772d94bdfbcd0f6bc91d9458ad5582",
    "pipeline": "master"
  }
]
```

**Comment by aaslamin** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "victoria-metrics",
    "component": "backend",
    "pr-id": "726",
    "commit-sha": "32d753a061cf5c7baa14689bbb56cb80b68b8d8e",
    "pipeline": "master"
  },
  {
    "project": "victoria-metrics",
    "component": "gaia",
    "pr-id": "471",
    "commit-sha": "ad180e323d772d94bdfbcd0f6bc91d9458ad5582",
    "pipeline": "master"
  }
]
```

**Comment by aaslamin** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "victoria-metrics",
    "component": "backend",
    "pr-id": "726",
    "commit-sha": "4f8ad9b8a02696e9db8ca58bbfcbc3b6bcebd510",
    "pipeline": "master"
  },
  {
    "project": "victoria-metrics",
    "component": "gaia",
    "pr-id": "471",
    "commit-sha": "ad180e323d772d94bdfbcd0f6bc91d9458ad5582",
    "pipeline": "master"
  }
]
```

**Comment by aaslamin** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "victoria-metrics",
    "component": "backend",
    "pr-id": "726",
    "commit-sha": "6342b110e7d7cea50c96bf5329183d473d614c1b",
    "pipeline": "master"
  },
  {
    "project": "victoria-metrics",
    "component": "gaia",
    "pr-id": "471",
    "commit-sha": "ad180e323d772d94bdfbcd0f6bc91d9458ad5582",
    "pipeline": "master"
  }
]
```

**Comment by aaslamin** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "victoria-metrics",
    "component": "backend",
    "pr-id": "726",
    "commit-sha": "072bfb2cbca54a7be399bb734b2cd6ca8be349ba",
    "pipeline": "master"
  },
  {
    "project": "victoria-metrics",
    "component": "gaia",
    "pr-id": "471",
    "commit-sha": "ad180e323d772d94bdfbcd0f6bc91d9458ad5582",
    "pipeline": "master"
  }
]
```

**Comment by aaslamin** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "victoria-metrics",
    "component": "gaia",
    "pr-id": "471",
    "commit-sha": "39654efca243c3d4fa082b63881ae8e2aa7a39db",
    "pipeline": "master"
  },
  {
    "project": "victoria-metrics",
    "component": "backend",
    "pr-id": "726",
    "commit-sha": "1a6f76673a83049face6d92216700da1214a4fa0",
    "pipeline": "master"
  }
]
```

**Comment by aaslamin** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "victoria-metrics",
    "component": "backend",
    "pr-id": "726",
    "commit-sha": "1a6f76673a83049face6d92216700da1214a4fa0",
    "pipeline": "master"
  },
  {
    "project": "victoria-metrics",
    "component": "gaia",
    "pr-id": "471",
    "commit-sha": "39654efca243c3d4fa082b63881ae8e2aa7a39db",
    "pipeline": "master"
  }
]
```

**Comment by aaslamin** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "victoria-metrics",
    "component": "backend",
    "pr-id": "726",
    "commit-sha": "1a6f76673a83049face6d92216700da1214a4fa0",
    "pipeline": "master"
  },
  {
    "project": "victoria-metrics",
    "component": "gaia",
    "pr-id": "471",
    "commit-sha": "39654efca243c3d4fa082b63881ae8e2aa7a39db",
    "pipeline": "master"
  }
]
```

**Comment by aaslamin** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "victoria-metrics",
    "component": "backend",
    "pr-id": "726",
    "commit-sha": "1a6f76673a83049face6d92216700da1214a4fa0",
    "pipeline": "master"
  },
  {
    "project": "victoria-metrics",
    "component": "gaia",
    "pr-id": "471",
    "commit-sha": "39654efca243c3d4fa082b63881ae8e2aa7a39db",
    "pipeline": "master"
  }
]
```

**Comment by aaslamin** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "victoria-metrics",
    "component": "backend",
    "pr-id": "726",
    "commit-sha": "e46144827ba950cac6906ccf63e1609525bca092",
    "pipeline": "master"
  },
  {
    "project": "victoria-metrics",
    "component": "gaia",
    "pr-id": "471",
    "commit-sha": "39654efca243c3d4fa082b63881ae8e2aa7a39db",
    "pipeline": "master"
  }
]
```

**Comment by aaslamin** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "victoria-metrics",
    "component": "backend",
    "pr-id": "726",
    "commit-sha": "1bc302c3b6995b987be89509d10a774eeb0d9d3c",
    "pipeline": "master"
  },
  {
    "project": "victoria-metrics",
    "component": "gaia",
    "pr-id": "471",
    "commit-sha": "39654efca243c3d4fa082b63881ae8e2aa7a39db",
    "pipeline": "master"
  }
]
```

File: specs/reportsquery.spec (hunk 5 above)
Comment (aaslamin) [reply specs/reportsquery.spec]: It's gone! \o/

**Review by primalmotion — APPROVED** [review]

**Comment by aaslamin** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "victoria-metrics",
    "component": "backend",
    "pr-id": "726",
    "commit-sha": "51a0e2596db048dd6d5e33a17fa7d65c540b1f7e",
    "pipeline": "master"
  },
  {
    "project": "victoria-metrics",
    "component": "gaia",
    "pr-id": "471",
    "commit-sha": "9fce2c17e68396892fe95bf3e3527a7e919b685a",
    "pipeline": "master"
  }
]
```

**Comment by aaslamin** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "victoria-metrics",
    "component": "backend",
    "pr-id": "726",
    "commit-sha": "60edbcc9401630d37522f64bd75529c887bdb0ed",
    "pipeline": "master"
  },
  {
    "project": "victoria-metrics",
    "component": "gaia",
    "pr-id": "471",
    "commit-sha": "9fce2c17e68396892fe95bf3e3527a7e919b685a",
    "pipeline": "master"
  }
]
```

**Comment by aaslamin** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "victoria-metrics",
    "component": "backend",
    "pr-id": "726",
    "commit-sha": "60edbcc9401630d37522f64bd75529c887bdb0ed",
    "pipeline": "master"
  },
  {
    "project": "victoria-metrics",
    "component": "gaia",
    "pr-id": "471",
    "commit-sha": "9fce2c17e68396892fe95bf3e3527a7e919b685a",
    "pipeline": "master"
  }
]
```

**Comment by aaslamin** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "victoria-metrics",
    "component": "backend",
    "pr-id": "726",
    "commit-sha": "036b4dfd28219acf88b11f5f5938b45852d5a6d4",
    "pipeline": "master"
  },
  {
    "project": "victoria-metrics",
    "component": "gaia",
    "pr-id": "471",
    "commit-sha": "9fce2c17e68396892fe95bf3e3527a7e919b685a",
    "pipeline": "master"
  },
  {
    "project": "victoria-metrics",
    "component": "voila",
    "pr-id": "118",
    "commit-sha": "abcc5faefcbb9a5fbb5991c7fcadfe2302a1e08a",
    "pipeline": "master"
  }
]
```

**Pull request merged**